src/trans.rs

```diff
@@ -250,20 +322,78 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
 
                         match *destination {
                             Some((ref lvalue, _)) => {
-                                let (lvalue_index, _) = self.trans_lval(lvalue);
+                                let dest = self.trans_lval(lvalue);
 
                                 if b_fnty == BinaryenNone() {
                                     // The result of the Rust call is put in MIR into a tmp local,
                                     // but the wasm function returns void (like the print externs)
                                     // TODO: what representation should the unit type have in wasm ?
```

Comment:
I'd say ideally the unit type should have no representation, which I guess means it's represented as a 0 byte value. If we have to give it a value, maybe 0 or -1 as an i32.